substrate/state-machine/src/testing.rs

```diff
@@ -17,48 +17,64 @@
 //! Test implementation for Externalities.
 
 use std::collections::HashMap;
+use std::cmp::Ord;
 use super::Externalities;
 use triehash::trie_root;
+use hashdb::Hasher;
+use rlp::Encodable;
+use std::marker::PhantomData;
 
-/// Simple HashMap based Externalities impl.
-pub type TestExternalities = HashMap<Vec<u8>, Vec<u8>>;
+/// Simple HashMap-based Externalities impl.
+pub struct TestExternalities<H> {
+	inner: HashMap<Vec<u8>, Vec<u8>>,
+	hasher: PhantomData<H>,
+}
+
+impl<H: Hasher> TestExternalities<H> {
+	#[cfg(test)]
```

Comment:
is this `#[cfg]` necessary?

Reply:
I think it's wrong. `TestExternalities` is exported (and re-exported again in `runtime-io`) so I think it needs to be `pub`. Good catch.

Reply:
…and this change will break things in other crates. I don't think this PR can be merged as-is.